pinc/DPage.inc

```diff
@@ -463,6 +463,7 @@ function Page_clearRound(string $projectid, string $image, Round $round, string
 
 // The numeric values of the following array must not change as they are
 // saved in the database to indicate the bad page reason.
+global $PAGE_BADNESS_REASONS;
```

Comment:
If I read [the php page on global](https://www.php.net/manual/en/language.variables.scope.php), this usage of `global` in the global scope is only useful if we want to include `pinc/DPPage.inc` inside a function. Is that something we want to support?

Reply:
Your reading is correct. The trick is how phpunit pulls in files is not always at the global context so tests fail because we didn't actually define the globals as globals.

Reply:
Because what I _think_ happens is that during some test the autoloader is being invoked from within a function, and the autoloader loads the file within the function, and that file may load other files (like `DPage.inc`) which is then loading the variable in the function's context rather than the global context.

Reply:
OK, that's a bit sad that we need to build this work-around for our tests when it's not something we wouldn't want to support. Thanks for updating those!

Reply:
Yeah. The solution is to stop using globals because it's a bad practice, but that's a ways out.